### PR TITLE
pkg/fileutil: support perallocate

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd/issues/2336

### CODE REVIEW

File: wal/wal.go [hunk 1]
```diff
@@ -203,6 +203,11 @@ func openAtIndex(dirpath string, snap walpb.Snapshot, write bool) (*WAL, error)
 			rc.Close()
 			return nil, err
 		}
+		err = fileutil.Preallocate(f, segmentSizeBytes)
+		if err != nil {
```
Comment (yichengq) [inline wal/wal.go]: `rc.Close()`? I see all previous error path does this.

**Comment by yichengq** [comment]
pretty clean one. :+1: 

Have you manually tested it? How is the performance going?

File: wal/wal.go (hunk 1 above)
Comment (xiang90) [inline wal/wal.go]: right we should

**Comment by xiang90** [comment]
@yichengq I tested it. I do not benchmark it at all.

**Comment by xiang90** [comment]
@yichengq The primary goal of this pr is 1 and 2 in https://github.com/coreos/etcd/issues/2336

**Comment by yichengq** [comment]
LGTM

**Pull request merged**